rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,146 @@
+        .layout_tests(true)
+        .merge_extern_blocks(true)
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        .prepend_enum_name(false)
+        .size_t_is_usize(true)
+        .sort_semantically(true)
+        // Structs
+        .allowlist_type(r#"^yp_\w+_node_t"#)
+        .allowlist_type("yp_buffer_t")
+        .allowlist_type("yp_comment_t")
+        .allowlist_type("yp_diagnostic_t")
+        .allowlist_type("yp_encoding_changed_callback_t")
+        .allowlist_type("yp_encoding_decode_callback_t")
+        .allowlist_type("yp_memsize_t")
+        .allowlist_type("yp_list_t")
+        .allowlist_type("yp_node_t")
+        .allowlist_type("yp_node_type")
+        .allowlist_type("yp_parser_t")
+        .allowlist_type("yp_pack_size")
+        // TODO: Commenting this because I can't figure out how to get bindgen to generate the
+        // inner-unions nicely. Hand-rolling this in src/lib.rs for now.
+        // .allowlist_type("yp_string_t")
+        .blocklist_type(r#"^yp_string_t\S*"#)
+        .allowlist_type("yp_string_list_t")
+        .allowlist_type("yp_token_type_t")
+        // Enums
+        .rustified_non_exhaustive_enum("yp_comment_type_t")
+        .rustified_non_exhaustive_enum("yp_context_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_indent_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_quote_t")
+        .rustified_non_exhaustive_enum("yp_lex_mode_t")
+        .rustified_non_exhaustive_enum("yp_lex_state_t")
+        .rustified_non_exhaustive_enum("yp_node_type")
+        .rustified_non_exhaustive_enum("yp_pack_encoding")
+        .rustified_non_exhaustive_enum("yp_pack_endian")
+        .rustified_non_exhaustive_enum("yp_pack_length_type")
+        .rustified_non_exhaustive_enum("yp_pack_result")
+        .rustified_non_exhaustive_enum("yp_pack_signed")
+        .rustified_non_exhaustive_enum("yp_pack_size")
+        .rustified_non_exhaustive_enum("yp_pack_type")
+        .rustified_non_exhaustive_enum("yp_pack_variant")
+        .rustified_non_exhaustive_enum("yp_token_type")
+        .rustified_non_exhaustive_enum("yp_unescape_type_t")
+        // Functions
+        .allowlist_function("yp_buffer_init")
+        .allowlist_function("yp_buffer_free")
+        .allowlist_function("yp_node_destroy")
+        .allowlist_function("yp_list_empty_p")
+        .allowlist_function("yp_list_free")
+        .allowlist_function("yp_list_init")
+        .allowlist_function("yp_node_memsize")
+        .allowlist_function("yp_pack_parse")
+        .allowlist_function("yp_parse")
+        .allowlist_function("yp_parse_serialize")
+        .allowlist_function("yp_parser_free")
+        .allowlist_function("yp_parser_init")
+        .allowlist_function("yp_parser_register_encoding_changed_callback")
+        .allowlist_function("yp_parser_register_encoding_decode_callback")
+        .allowlist_function("yp_prettyprint")
+        .allowlist_function("yp_regexp_named_capture_group_names")
```

Comment:
I don't think this needs to be exposed, i.e. I think the name are already exposed in the resulting AST nodes